src/python/tests/test_taxon_selector.py

```diff
@@ -0,0 +1,143 @@
+)
+
+@pytest.mark.parametrize("db", [{"src": "ncbi_db"}], indirect=True)
+class TestTaxonomySelection:
+    """Tests functions `~ensembl_compara.taxonomy.taxon_selector` in taxon_selector.py
+
+    Attributes:
+        dbc (DBConnection): Database connection to the unit test database.
+        dir (ptest.files_dir): Test files directory.
+
+    """
+
+    dbc = None # type: UnitTestDB
+    dir = None
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self, db: UnitTestDB) -> None:
+        """Loads the required fixtures and values as class attributes.
+
+        Args:
+            db: Generator of unit test database (fixture).
+        """
+        type(self).dbc = db.dbc
+        # pylint: disable=no-member
+        type(self).dir = pytest.files_dir / "ref_taxa" # type: ignore[attr-defined]
+
+    @pytest.mark.parametrize("exp_out1, exp_out2", [(["mammalia", "vertebrata"], [])])
+    def test_collect_taxonomys_from_path(self, exp_out1: list, exp_out2: list) -> None:
+        """Tests :func:`collect_taxonomys_from_path()`
+
+        Args:
+            exp_out1: Test directory with subdirectories named by existing taxa
+            exp_out2: Test directory with subdirectories named by random non-taxa
+        """
+        with self.dbc.session_scope() as session:
+            result = collect_taxonomys_from_path(session, str(self.dir))
+            # pylint: disable=no-member
+            null_result = collect_taxonomys_from_path(session, pytest.files_dir) # type: ignore[attr-defined]
+            assert result == exp_out1
+            assert null_result == exp_out2
+
+    @pytest.mark.parametrize(
+        "exp_out1, exp_out2, exp_out3, expectation",
+        [("mammalia", "vertebrata", "default", raises(NoResultFound))],
+    )
+    def test_match_taxon_to_reference(
+        self, exp_out1: str, exp_out2: str, exp_out3: str, expectation: ContextManager
+    ) -> None:
+        """Tests :func:`match_taxon_to_reference()`
+
+        Args:
+            exp_out1: A child clade of larger clade in taxon list
+            exp_out2: A valid taxon clade
+            exp_out3: A valid taxon clade
+            expectation: A random string
+        """
+        list1 = ["vertebrata", "mammalia", "eukaryota"]
+        list2 = ["cake", "chocolate", "bananna"]
+        with self.dbc.session_scope() as session:
+            result1 = match_taxon_to_reference(session, "Canis lupus", list1)
+            result2 = match_taxon_to_reference(session, "Gallus gallus", list1)
+            result3 = match_taxon_to_reference(session, "Canis lupus", list2)
+            assert result1 == exp_out1
+            assert result2 == exp_out2
+            assert result3 == exp_out3
+            with expectation:
+                result4 = match_taxon_to_reference(session, "cake", list2)
+                assert result4 == exp_out1
+
+    @pytest.mark.parametrize(
+        "taxon_1, fake_taxon, expectation", [("mammalia", "cake", "mammalia")]
```

Comment:
maybe adding tests that include `None`, `empty` strings to force unexpected outcomes and exception raises.
This comment also applies to other tests in this file.